JS/Lesson15/goksel/script.js

```diff
@@ -103,62 +104,66 @@ const stringExercise14 =
 //   stringExercise14.indexOf('searchString'), -1)
 // );
-
+console.log("Exercises14: " ,stringExercise14.substring(0, stringExercise14.indexOf("are") - 1) );
 console.log('------------------------');
 // 15. Remove extra spaces from the string stringExercise15 and print the result to the console.
 const stringExercise15 = '    Hello world    ';
 
-console.log('Exercise15: ', stringExercise15, stringExercise15.trim());
+console.log('Exercise15: ',stringExercise15.trim());
 console.log('------------------------');
 // 16. Extract the price and currency from a string below using "slice" method and print it to console.
 const planPriceString = 'Premium plan - 9.99 USD/month.'; // Should return "9.99 USD"
 
-console.log('Exercise16: ' /* Result here */);
+console.log('Exercise16: ', planPriceString.slice(planPriceString.indexOf('-') + 1, planPriceString.indexOf('/') ) );
 console.log('------------------------');
 
 // 17. Use the `repeat` method to create a string of asterisks (*) that forms a horizontal line with a length of 10 and print it to the console.
 
-console.log('Exercise17: ', /* Result here */);
+console.log('Exercise17: ', "*".repeat(10));
 
 console.log('------------------------');
 // 18. Use the `concat` method to combine three separate strings into one and print the result to the console.
 const string1Exercise18 = 'Hello, ';
 const string2Exercise18 = 'world';
 const string3Exercise18 = '!';
-console.log('Exercise18: ', /* Result here */);
+console.log('Exercise18: ',string1Exercise18.concat(string2Exercise18, string3Exercise18) );
 
 console.log('------------------------');
 // 19. Use the `split` and `join` methods to reverse the words in the string stringExercise28 and print the result to the console.
 const stringExercise19 = 'This is a sample sentence.';
-console.log('Exercise19: ', /* Result here */);
+console.log('Exercise19: ', stringExercise19.split(" ").reverse().join(" ") );
 
 console.log('------------------------');
 
 // 20. Use the `charAt` method to extract the first and last characters of a string and concatenate them into a new string. Then use the `toUpperCase` method to make the new string uppercase and print the result to the console.
 const stringExercise20 = 'JavaScript';
-
-console.log('Exercise20: ', /* Result here */);
+const first = stringExercise20.charAt(0);
+const last = stringExercise20.charAt(stringExercise20.length - 1);
+const result= first + last ;
+console.log('Exercise20: ', result.toUpperCase());
 
 console.log('------------------------');
 // 21. Use the `substring` method to extract the substring "quick brown fox" from the string stringExercise32 based on two indices and print it to the console.
 const stringExercise21 = 'The quick brown fox jumps over the lazy dog.';
 
-console.log('Exercise21: ', extractedSubstringExercise32);
+console.log('Exercise21: ', stringExercise21.substring(stringExercise21.indexOf("q"), stringExercise21.indexOf("jumps") -1) );
```

Comment:
This is good but could be dangerous as many words starts with "q".  This could be safer:
```
stringExercise21.indexOf("quick brown fox")
```

For the last index: the following word may be something other than "jumps" however the length of the "quick brown fox" will not change. You can base the logic on the start index and the length of the string you search for :)

Reply:
console.log('Exercise21: ', stringExercise21.substring(stringExercise21.indexOf("quick brown fox"),stringExercise21.indexOf("quick brown fox") + ('quick brown fox').length));